kentikapi/models/cloud_export.go

```diff
@@ -0,0 +1,239 @@
+	AWSProperties *AWSProperties
+	// AzureProperties are specific to Azure exports.
+	AzureProperties *AzureProperties
+	// GCEProperties are specific to Google Cloud export.
+	GCEProperties *GCEProperties
+	// IBMProperties are specific to IBM Cloud exports.
+	IBMProperties *IBMProperties
+	// BGPProperties are optional BGP related settings.
+	BGP *BGPProperties
+
+	// Read-only properties
+
+	// ID is the internal cloud export identifier. This is read-only and assigned by Kentik.
+	ID ID
+	// CurrentStatus is the most current status Kentik has about this export. This is read-only and assigned by Kentik.
+	CurrentStatus *CloudExportStatus
+}
+
+// AWSProperties are specific to Amazon Web Services VPC flow logs exports.
+type AWSProperties struct {
+	// Bucket is source S3 bucket to fetch VPC flow logs from.
+	Bucket string
+	// IAMRoleARN is an ARN for the IAM role to assume when fetching data or making AWS calls for this export.
+	IAMRoleARN string
+	// Region is an AWS region where the source S3 bucket resides.
+	Region string
+	// DeleteAfterRead enables the deletion of VPC flow log chunks from S3 after they have been read.
+	DeleteAfterRead *bool
+	// MultipleBuckets enables using multiple source S3 buckets.
+	MultipleBuckets *bool
+}
+
+// AzureProperties are specific to Azure exports.
+type AzureProperties struct {
+	// Location is an Azure location.
+	Location string
+	// ResourceGroup is the name of resource group from which to collect flow logs.
+	ResourceGroup string
+	// StorageAccount is the name of storage account for storing flow logs.  where flow logs will be collected.
+	StorageAccount string
+	// SubscriptionID is the Azure subscription ID.
+	SubscriptionID string
+	// SecurityPrincipalEnabled enables security principal.
+	SecurityPrincipalEnabled *bool
+}
+
+// GCEProperties are specific to Google Cloud export.
+type GCEProperties struct {
+	// Project is a GCE project name.
+	Project string
+	// Subscription is a GCE subscription name.
+	Subscription string
+}
+
+// IBMProperties are specific to IBM Cloud exports.
+type IBMProperties struct {
+	// Bucket is an IBM bucket.
+	Bucket string
+}
+
+// BGPProperties are optional BGP related settings.
+type BGPProperties struct {
+	// ApplyBGP enables applying BGP data discovered via another device to the flow from this export.
+	ApplyBGP *bool
+	// UseBGPDeviceID specifies which other device to get BGP data from.
+	UseBGPDeviceID string
+	DeviceBGPType  string
+}
+
+// CloudExportStatus is export task status.
+type CloudExportStatus struct {
+	Status string
+	// ErrorMessage holds current error information.
+	ErrorMessage string
+	// FlowFound informs whether flow logs were found.
+	FlowFound            *bool
+	APIAccess            *bool
+	StorageAccountAccess *bool
+}
+
+// CloudExportAWSRequiredFields is subset of fields required to create an AWS CloudExport.
+type CloudExportAWSRequiredFields struct {
+	Name          string
```

Comment:
Why not use a common struct for common required attributes and rely on [promoted struct fields](https://golangdocs.com/structs-in-golang)? Something like this:

```go
type CommonRequiredFields struct {
    Name    string
    PlanID  string
}

type CloudExportAWSRequiredFields struct {
    CommonRequiredFields
    AWSProperties AWSPropertiesRequiredFields
}
```